workflow/lib/shared/target_utils.py

```diff
@@ -1,7 +1,22 @@
 """Helper functions for using Snakemake outputs and targets for use with Brieflow."""
 
+import string
+from pathlib import Path
 from snakemake.io import expand, temp, ancient
+import itertools
 
+def clean_value(val):
+    """Convert numpy types to Python native types.
+    
+    Args:
+        val: Value to clean (could be numpy type or native Python type)
+        
+    Returns:
+        Native Python type (int, str, etc.)
+    """
+    if hasattr(val, 'item'):  # Check if it's a numpy type
+        return val.item()
+    return val
```

Comment:
Consider type casting inside code without this function which will give us more clarity on how things are getting cleaned.